dl/src/main/scala/com/intel/analytics/bigdl/models/rnn/Train.scala

```diff
@@ -48,38 +48,46 @@ object Train {
 
       val (trainSet, validationSet, dictionaryLength) = if (!sc.isDefined) {
         val logData = Source.fromFile(param.dataFolder + "/" + "train.txt").getLines().toArray
-        val tokens = DataSet.array(logData.filter(!_.isEmpty))
-          .transform(DocumentTokenizer())
+        val trainSents = DataSet.array(logData
+          .filter(!_.isEmpty)).transform(SentenceSplitter())
+        val output_sents = trainSents.toLocal().data(train = false).flatMap(item => item.iterator)
+        val tokens = DataSet.array(output_sents.toArray).transform(DocumentTokenizer())
         val dictionary = Dictionary(tokens.toLocal().data(false), param.vocabSize)
         dictionary.save(param.saveFolder)
         val isTable = if (param.batchSize > 1) true else false
         println("vocabulary size = " + dictionary.vocabSize())
+        val valData = Source.fromFile(param.dataFolder + "/" + "val.txt").getLines().toArray
+        val valSents = DataSet.array(valData
+          .filter(!_.isEmpty)).transform(SentenceSplitter())
+        val valoutput = valSents.toLocal().data(train = false).flatMap(item => item.iterator)
```

Comment:
This will be fine if the ArrayBuffer is called each iteration in the SentenceSplitter(). However, if you define ArrayBuffer once in the constructor and reuse it, the output data here will be the final sample instead of all the samples.